Assets/MixedRealityToolkit.Services/InputSimulation/Implementation/InputSimulationService.cs

```diff
@@ -0,0 +1,430 @@
+            {
+                handDataProvider = null;
+            }
+        }
+
+        private Vector3 lastMousePosition;
+        private bool wasFocused;
+        private bool wasCursorLocked;
+
+        private void ResetMouseDelta()
+        {
+            lastMousePosition = UnityEngine.Input.mousePosition;
+        }
+
+        private MouseDelta UpdateMouseDelta()
+        {
+            var profile = InputSimulationProfile;
+
+            bool isFocused = Application.isFocused;
+            bool gainedFocus = (!wasFocused && isFocused);
+            wasFocused = isFocused;
+
+            bool isCursorLocked = UnityEngine.Cursor.lockState != CursorLockMode.None;
+            bool cursorLockChanged = (wasCursorLocked != isCursorLocked);
+            wasCursorLocked = isCursorLocked;
+
+            // Reset in cases where mouse position is jumping
+            if (gainedFocus || cursorLockChanged)
+            {
+                ResetMouseDelta();
+                return new MouseDelta();
+            }
+            else
+            {
+                // Use frame-to-frame mouse delta in pixels to determine mouse rotation.
+                // The traditional GetAxis("Mouse X") method doesn't work under Remote Desktop.
+                Vector3 screenDelta;
+                Vector3 worldDelta;
+                if (UnityEngine.Cursor.lockState == CursorLockMode.Locked)
+                {
+                    screenDelta.x = UnityEngine.Input.GetAxis(profile.MouseX);
+                    screenDelta.y = UnityEngine.Input.GetAxis(profile.MouseY);
+
+                    worldDelta.z = UnityEngine.Input.GetAxis(profile.MouseScroll);
+                }
+                else
+                {
+                    screenDelta.x = (UnityEngine.Input.mousePosition.x - lastMousePosition.x);
+                    screenDelta.y = (UnityEngine.Input.mousePosition.y - lastMousePosition.y);
+
+                    worldDelta.z = UnityEngine.Input.mouseScrollDelta.y;
+                }
+
+                // Interpret scroll values as world space delta
+                worldDelta.z *= profile.HandDepthMultiplier;
+
+                // Convert world space scroll delta into screen space pixels
+                screenDelta.z = WorldToScreen(new Vector2(worldDelta.z, 0)).x;
+
+                // Convert screen space x/y delta into world space
+                Vector2 worldDelta2D = ScreenToWorld(new Vector2(screenDelta.x, screenDelta.y));
+                worldDelta.x = worldDelta2D.x;
+                worldDelta.y = worldDelta2D.y;
+
+                lastMousePosition = UnityEngine.Input.mousePosition;
+
+                return new MouseDelta()
+                {
+                    screenDelta = screenDelta,
+                    worldDelta = worldDelta,
+                    viewportDelta = CameraCache.Main.ScreenToViewportPoint(screenDelta),
+                };
+            }
+        }
+
+        private const float MouseWorldDepth = 0.5f;
```

Comment:
>private const float MouseWorldDepth = 0.5f; [](start = 8, length = 43)

What's the significance of the .5 number? (and is it designed to also match the other .5f values down below?)

Reply:
Added comments
```cpp
        // Default world-space distance for converting screen/viewport scroll offsets into world space depth offset.
        // The pixel-to-world-unit ratio changes with depth, so have to chose a fixed distance for conversion.
        private const float mouseWorldDepth = 0.5f;
        // Center of the viewport is at (0.5, 0.5)
        private readonly Vector2 viewportCenter = new Vector2(0.5f, 0.5f);
```